extensions/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/manager/DataPlaneManagerImpl.java

```diff
@@ -109,7 +118,15 @@ private void run() {
                     continue;
                 }
                 final var polledRequest = request;
-                pipelineService.transfer(request).whenComplete((result, exception) -> {
+
+                TransferService transferService = resolveTransferService(polledRequest);
+                if (transferService == null) {
+                    // Should not happen since resolving a transferService is part of payload validation
```

Comment:
Should't we throw an exception here?

Reply:
Added conditional block to avoid NPE. We don't want to raise an exception since that would cause the transfer to be picked up again. We do want to record the transfer as in error, but that feature is not implemented (viz. line 132)